woodwork/tests/utils/test_concat.py

```diff
@@ -386,28 +386,6 @@ def test_concat_cols_validate_schema(mock_validate_accessor_params, sample_df):
     assert mock_validate_accessor_params.called
 
 
-def test_concat_cols_mismatched_index_adds_single_nan(sample_df):
-    sample_df.ww.init(index="id")
```

Comment:
If I'm understanding correctly, this test would now not pass because introducing a nan into the resulting dataframe that has an index is not allowed. 

Can we remove this `index='id'` and keep the test? I think doing that shouldn't change the usefulness of this test, which is mostly about the behavior when `concat_columns` introduces nans in non nullable logical types

Reply:
Yep sounds good